apps/cli/src/auth-server.ts

```diff
@@ -0,0 +1,84 @@
+import dotenv from 'dotenv';
+dotenv.config();
+import express from 'express';
+import fetch from 'node-fetch';
+import type { Request, Response } from 'express';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+const CLIENT_ID = process.env.GITHUB_CLIENT_ID || '';
+const CLIENT_SECRET = process.env.GITHUB_CLIENT_SECRET || '';
+const PORT = process.env.PORT || 3456;
+const CALLBACK_URL = process.env.GITHUB_CALLBACK_URL || `http://localhost:${PORT}/api/auth/github/callback`;
+
+const TOKEN_PATH = path.join(os.homedir(), '.open-swe-cli', 'github_token.json');
+
+let accessToken: string | null = null;
+
+function saveToken(tokenData: any) {
+  const dir = path.dirname(TOKEN_PATH);
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir, { recursive: true, mode: 0o700 });
+  }
+  fs.writeFileSync(TOKEN_PATH, JSON.stringify(tokenData, null, 2), { mode: 0o600 });
+}
+
+function loadToken() {
+  if (fs.existsSync(TOKEN_PATH)) {
+    return JSON.parse(fs.readFileSync(TOKEN_PATH, 'utf8'));
+  }
+  return null;
+}
+
+const app = express();
+
+// 1. Start OAuth flow
+app.get('/api/auth/github/login', (_req: Request, res: Response) => {
+  const state = Math.random().toString(36).substring(2);
+  const githubAuthUrl = `https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&redirect_uri=${encodeURIComponent(CALLBACK_URL)}&state=${state}`;
+  res.redirect(githubAuthUrl);
+});
+
+// 2. Handle OAuth callback
+app.get('/api/auth/github/callback', async (req: Request, res: Response) => {
+  const code = req.query.code as string;
+  // Optionally validate state here
+  if (!code) {
+    return res.status(400).send('Missing code parameter');
+  }
+  // Exchange code for access token
+  const tokenRes = await fetch('https://github.com/login/oauth/access_token', {
+    method: 'POST',
+    headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      client_id: CLIENT_ID,
+      client_secret: CLIENT_SECRET,
+      code,
+      redirect_uri: CALLBACK_URL,
+    }),
+  });
+  const tokenData = await tokenRes.json();
+  if (tokenData.error) {
```

Comment:
`.ok` will be true if the fetch res is between 200-299. good to include (although your `.error` check might be doing the same thing?
```suggestion
  const tokenData = await tokenRes.json();
  if (tokenData.error || !tokenRes.ok) {
```